src/lib/video-progress-reporting.ts

```diff
@@ -0,0 +1,58 @@
+import { once } from 'lodash-es';
+import { getAdvertById } from './dfp/get-advert-by-id';
+import { checkConsent as checkConsentForReporting } from './send-commercial-metrics';
+
+const endpoint = window.guardian.config.page.isDev
+	? '//logs.code.dev-guardianapis.com/log'
+	: '//logs.guardianapis.com/log';
+
+// This is a map of video progress for each slotId.
+const videoProgress: Record<string, number | null> = {};
+
+const sendProgress = once(() => {
+	Object.entries(videoProgress).forEach(([key, value]) => {
+		const advert = getAdvertById(key);
+
+		if (!advert?.creativeId || !advert.lineItemId) {
+			return;
+		}
+		const { creativeId, lineItemId } = advert;
+
+		const progress = value ?? 0;
+
+		void fetch(endpoint, {
+			method: 'POST',
+			body: JSON.stringify({
+				label: 'commercial.interscroller.videoProgress',
```

Comment:
changing this will break any current video reports so it'll have to keep interscroller in the label!